lib/kubernetes-deploy/renderer.rb

```diff
@@ -46,13 +47,14 @@ def render_template(filename, raw_template)
       raise InvalidTemplateError.new(err.message, filename: filename, content: raw_template)
     end
 
-    def render_partial(partial, locals)
+    def render_partial(template_dir, partial, locals)
```

Comment:
This PR adds `template_dir` to the method signature of `render_partial, find_partial, TemplateContext#new, and TemplateContext#partial`. It's annoying to have to pass this context around to so many places, but it seems unavoidable. Interested to hear any alternatives anyone may have.